ecl/eclagent/eclagent.cpp

```diff
@@ -1953,11 +1936,17 @@ void EclAgent::doProcess()
     DBGLOG("Process complete");
     // Add some timing stats
     bool deleteJobTemps = true;
+    WorkunitUpdate w(nullptr);
     try
     {
         updateWULogfile();//Update workunit logfile name in case of date rollover
 
-        WorkunitUpdate w = updateWorkUnit();
+        //This is before w is initialised, so that restoreCluster can update the workunit.
+        //It cannot take w as an argument, because restoreCluster is declared in IGlobalCodeContext with no parameters
```

Comment:
to avoid the double lock, would be better to have a new restoreCluster(IWorkUnit *) method that this called after establishing 'w', and the IGlobalCodeContext::restoreCluster() call could estalish lock and call it.

Reply:
@nhalliday - above comments no longer apply, should remove.